standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/dataconnector/jdbc/AbstractJDBCConnectorProvider.java

```diff
@@ -185,9 +185,18 @@ protected Connection getConnection() {
     return null;
   }
 
-  protected abstract ResultSet fetchTableMetadata(String tableName) throws MetaException;
+  protected ResultSet fetchTableMetadata(String tableName) throws MetaException {
+    try {
+      return fetchTablesViaDBMetaData(tableName);
+    }
+    catch (SQLException sqle) {
+      throw new MetaException("Error while trying to access the table names in the database" + sqle);
+    }
+  }
 
-  protected abstract ResultSet fetchTableNames() throws MetaException;
+  protected ResultSet fetchTableNames() throws MetaException {
+    return fetchTableMetadata(null);
+  }
```

Comment:
I don't see any usages of `fetchTableMetadata` or `fetchTableNames` in the project. How about removing these methods completely?